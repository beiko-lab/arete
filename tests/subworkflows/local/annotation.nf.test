nextflow_workflow {

    name "Test Workflow ANNOTATE_ASSEMBLIES"
    script "subworkflows/local/annotation.nf"
    workflow "ANNOTATE_ASSEMBLIES"

    test("Annotation subworkflow runs without failures") {

        when {
            params {
                // define parameters here. Example:
                outdir = "$outputDir"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = [[id:'SRR14022735'], '$baseDir/test/SRR14022735_T1.scaffolds.fa']
                input[1] = []
                input[2] = []
                input[3] = []
                input[4] = []
                input[5] = []
                input[6] = []
                """
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 13
            assert workflow.out.gff.size() == 1
            assert workflow.out.gff.get(0).get(1) ==~ ".*/SRR14022735.gff"
        }

    }

}
