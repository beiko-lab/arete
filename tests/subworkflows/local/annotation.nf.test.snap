{
    "Should run without failures": {
        "content": [
            {
                "multiqc": [
                    [
                        "SRR14022735.txt:md5,816c0b42cee404e51bb572d080d57e2d"
                    ]
                ],
                "gff": [
                    [
                        {
                            "id": "SRR14022735"
                        },
                        "SRR14022735.gff:md5,024ead71b61e5a57daf904cfd782fa4f"
                    ]
                ],
                "annotation_software": [
                    "card.version.txt:md5,aac12a38555b613763f9a8a0bb01b947",
                    "mob.version.txt:md5,51f907f278a82a9f2fc1125dc528c514",
                    "rgi.version.txt:md5,0c324d0f9ff6c20c87e11fdb36877197",
                    "versions.yml:md5,72f6cd794b597ebd14c38b28dd879e27",
                    "versions.yml:md5,35bc3fab77dd594ca553b7652ec0c623"
                ]
            }
        ],
        "timestamp": "2023-02-07T18:32:11+0000"
    }
}